tests/test_max_nfev.py

```diff
@@ -84,17 +84,17 @@ def test_max_nfev_Model(modelGaussian, minimizerGaussian, method):
                          [('leastsq', 2000*(nvarys+1)),
                           ('least_squares', 1000*(nvarys+1)),
                           ('nelder', 1000*(nvarys+1)),
-                          ('brute', np.inf),
                           ('ampgo', 1000000*(nvarys+1)),
-                          ('basinhopping', 2000*(nvarys+1)),
+                          # ('brute', np.inf),
+                          # ('basinhopping', 2000*(nvarys+1)),
                           ('differential_evolution', 1000*(nvarys+1)),
                           ('shgo', 1000000*(nvarys+1)),
                           ('dual_annealing', 1.e7)])
 def test_default_max_nfev(modelGaussian, minimizerGaussian, method,
                           default_max_nfev):
     """Test the default values when setting max_nfev=None."""
-    if method in ('brute', 'basinhopping'):
-        pytest.xfail(f'max_nfev not yet supported in {method}')  # FIXME
+    # if method in ('brute', 'basinhopping'):
+    #     pytest.xfail(f'max_nfev not yet supported in {method}')  # FIXME
```

Comment:
I prefer to keep this as it was; `xfail` is exactly intended for this purpose. The tests should pass, but currently do not since it's not implemented in these methods. Leaving it in there reminds us that we *should* fix this; commenting it out doesn't help or change anything useful [ in my opinion of course ;) ]

Reply:
OK, I can put that back in.   I don't expect it to ever change, though, as neither of the `scipy.optimize` functions have a "max number of function calls" parameter.

Reply:
true, but since we use `max_nfev` now internally in the `__residual` function to decide if we want to abort the fit, it could be supported these methods as well - correct?

Reply:
but then why are those tests marked as "expected to fail"?

Reply:
because the `max_nfev` is not added in the `basinhopping` and `brute` methods in `minimizer.py`. At the time that you added this general way of specifying the maximum number of function evaluations I tried to add it, but couldn't quickly get it to behave correctly (some tests started to fail). If I remember correctly we decided back then to merge it since it worked for most of the methods, with the idea of getting back to this and fix it at a later time.